Utils/Campaign builder/Unit1.pas

```diff
@@ -363,6 +363,12 @@ procedure TForm1.RefreshFlags;
     //Position node centers, so that if someone changes the nodes they still look correct
     imgNodes[I].Left := Image1.Left + C.Maps[fSelectedMap].Nodes[I].X - imgNodes[I].Width div 2;
     imgNodes[I].Top := Image1.Top + C.Maps[fSelectedMap].Nodes[I].Y - imgNodes[I].Height div 2;
+    //Nodes If the position is less than zero
+    //We are making it equal to zero
+    if imgNodes[I].Left < 0 then
+      imgNodes[I].Left := 0;
+    if imgNodes[I].Top < 0 then
+      imgNodes[I].Top := 0;

```

Comment:
Maybe you should use EnsureRange here instead like we do in NodeMove? Otherwise the nodes could be positioned off the bottom/right edge of the image. And what about the flags, should we EnsureRange when positioning them in this procedure?

Also, a problem with this implementation is that the change does not get written back to the campaign data. So if the node is positioned at 0 in the campaign data, in the builder UI here we will set it to imgNodes[I].Width div 2 so it's not off screen, but when you save the campaign file will still have it at 0 and partially off screen.